pkg/http/api.go

```diff
@@ -39,7 +39,7 @@ type api struct {
 	secretStores          map[string]secretstores.SecretStore
 	json                  jsoniter.API
 	actor                 actors.Actors
-	pubSub                pubsub.PubSub
+	publishFn             func(req *pubsub.PublishRequest) error
```

Comment:
how about defining type "PublishFunction" for " func(req *pubsub.PublishRequest) error " and use it..?

Reply:
See response below